### PR TITLE
Yc/add constant diff model

### PR DESCRIPTION
**Pull request by yairchn**
testing our ability to use a different turn_conv model. I still have a few issues left

### CODE REVIEW

File: driver/callbacks.jl [hunk 1]
```diff
@@ -37,11 +37,13 @@ function affect_io!(integrator)
...
 
 function affect_filter!(integrator)
-    UnPack.@unpack edmf, grid, param_set, aux, case = integrator.p
+    UnPack.@unpack turb_conv, grid, param_set, aux, case = integrator.p
     t = integrator.t
-    state = TC.State(integrator.u, aux, integrator.du)
+    state = TC.State(integrator.u, aux, ODE.get_du(integrator))
     surf = get_surface(case.surf_params, grid, state, t, param_set)
-    TC.affect_filter!(edmf, grid, state, param_set, surf, case.casename, t)
+    if turb_conv isa TC.EDMFModel
```
Comment (yairchn) [inline driver/callbacks.jl:44]: @charleskawczynski  is there a better way to do this ?

File: driver/callbacks.jl [hunk 2]
```diff
@@ -100,8 +102,43 @@ function dt_max!(integrator)
     ODE.u_modified!(integrator, false)
 end
 
+function diffusivity_dt_max!(integrator)
```
Comment (yairchn) [inline driver/callbacks.jl:105]: this requires some planning

File: driver/dycore.jl [hunk 3]
```diff
@@ -197,75 +204,190 @@ function satadjust(param_set::APS, grid, state)
...
-    UnPack.@unpack edmf, precip_model, grid, param_set, case, aux, TS = params
+    UnPack.@unpack turb_conv, precip_model, grid, param_set, case, aux, TS = params
 
     state = TC.State(prog, aux, tendencies)
 
     Δt = TS.dt
     surf = get_surface(case.surf_params, grid, state, t, param_set)
     force = case.Fo
     radiation = case.Rad
+    if turb_conv isa TC.EDMFModel
```
Comment (yairchn) [inline driver/dycore.jl:218]: is there a better way to do this ?

File: driver/dycore.jl [hunk 4]
```diff
@@ -197,75 +204,190 @@ function satadjust(param_set::APS, grid, state)
...
+end
+
+function update_aux!(turb_conv::TC.DiffusivityModel, grid, state, surf, param_set, t, Δt)
+    kc_surf = TC.kc_surface(grid)
+    FT = eltype(grid)
+    aux_gm_f = TC.face_aux_grid_mean(state)
+    prog_gm = TC.center_prog_grid_mean(state)
+    wvec = CC.Geometry.WVector
+    ∇ = CCO.DivergenceC2F(; bottom = CCO.SetDivergence(FT(0)), top = CCO.SetDivergence(FT(0)))
+    # ∇u = ∇(wvec(prog_gm.u))
```
Comment (yairchn) [inline driver/dycore.jl:262]: I still need to compute the shear here

File: driver/callbacks.jl (hunk 1 above)
Comment (charleskawczynski) [reply driver/callbacks.jl:45]: Yes, rename the function to `affect_filter!` edmf and only add this to the callback set when we're using EDMF. We probably shouldn't be running through this function at all if we don't have edmf.

**Comment by charleskawczynski** [comment]
@yairchn, okay to close?

**Comment by yairchn** [comment]
yes

**Pull request closed without merging**